### PR TITLE
Implement Reset for MySQL 

### PR DESCRIPTION
**Pull request by longquanzheng**

### CODE REVIEW

File: common/persistence/sql/sqlExecutionManager.go [hunk 1]
```diff
@@ -622,8 +624,217 @@ func updateBufferedEvents(tx sqldb.Tx, batch *p.DataBlob, clear bool, shardID in
...
+				}
+			}
+
+			if err := updateExecution(tx, currExecutionInfo, currReplicationState, shardID); err != nil {
+				return &workflow.InternalServiceError{
+					Message: fmt.Sprintf("ResetWorkflowExecution operation failed. Failed to update executions row. Erorr: %v", err),
+				}
+			}
+		} else {
+			if err := lockAndCheckNextEventID(tx, shardID, domainID, workflowID, currRunID, request.Condition); err != nil {
```
Comment (venkat1109) [inline common/persistence/sql/sqlExecutionManager.go]: can you add some comments on why these locks need to be acquired / what is base/current/updateCurr etc ?

File: schema/mysql/v57/cadence/schema.sql [hunk 2]
```diff
@@ -189,8 +191,13 @@ CREATE TABLE replication_tasks (
 	first_event_id BIGINT NOT NULL,
 	next_event_id BIGINT NOT NULL,
 	version BIGINT NOT NULL,
-  last_replication_info BLOB NOT NULL,
+    last_replication_info BLOB NOT NULL,
```
Comment (venkat1109) [inline schema/mysql/v57/cadence/schema.sql]: fix formatting

File: common/persistence/sql/workflowStateMaps.go [hunk 3]
```diff
@@ -650,13 +650,15 @@ func updateBufferedReplicationTasks(tx sqldb.Tx,
...
-			FirstEventID: newBufferedReplicationTask.FirstEventID,
-			Version:      newBufferedReplicationTask.Version,
-			NextEventID:  newBufferedReplicationTask.NextEventID,
+			ShardID:                 int64(shardID),
+			DomainID:                domainID,
+			WorkflowID:              workflowID,
+			RunID:                   runID,
+			FirstEventID:            newBufferedReplicationTask.FirstEventID,
+			Version:                 newBufferedReplicationTask.Version,
+			NextEventID:             newBufferedReplicationTask.NextEventID,
```
[the remark this answers is not in the document]
Comment (longquanzheng) [reply common/persistence/sql/workflowStateMaps.go]: Yes we have pretty good coverage for the new fields for eventsV2: https://github.com/uber/cadence/blob/master/common/persistence/persistence-tests/executionManagerTestForEventsV2.go#L573

File: common/persistence/sql/sqlExecutionManager.go (hunk 1 above)
Comment (longquanzheng) [reply common/persistence/sql/sqlExecutionManager.go]: I already verified and I failed the test when I forgot to add those fields

File: schema/mysql/v57/cadence/schema.sql (hunk 2 above)
Comment (longquanzheng) [reply schema/mysql/v57/cadence/schema.sql]: sounds good

**Comment by longquanzheng** [comment]
I will work on enabling eventsV2 test after this PR: https://github.com/uber/cadence/issues/1686

**Pull request merged**